client/state/current-user/email-verification/actions.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_VERIFY_REQUEST,
+	EMAIL_VERIFY_REQUEST_SUCCESS,
+	EMAIL_VERIFY_REQUEST_FAILURE,
+	EMAIL_VERIFY_STATE_RESET,
+} from 'state/action-types';
+import wpcom from 'lib/wp';
+
+export function verifyEmail() {
```

Comment:
@gwwar I updated this PR to use the data layer for sending an email verification in 053cee7a0e106ca19b8e44974ea727d654daaba9 Let me know what you think :)